key.go

```diff
@@ -459,6 +474,12 @@ func (k Key) validate(op KeyOp) error {
 		if crv == CurveInvalid || (len(x) == 0 && len(d) == 0) {
 			return ErrInvalidKey
 		}
+		if len(x) > 0 && len(x) != ed25519.PublicKeySize {
+			return fmt.Errorf("invalid x size: expected %d, got %d", ed25519.PublicKeySize, len(x))
+		}
+		if len(d) > 0 && len(d) != ed25519.SeedSize {
+			return fmt.Errorf("invalid d size: expected %d, got %d", ed25519.SeedSize, len(d))
```

Comment:
I'm wondering whether it'd be worth factoring out this pattern into its own InvalidSizeError?

Reply:
Done. I've refactored a bit `Key.validate` to consolidate some error messages. Haven't implemented `InvalidSizeError`, adding new error types increase the API surface of our library, which means more chances to break compatibility. We already have a `ErrInvalidKey` exported error, so I've reused it and added some more context.